File: ruby/easy/symmetric_tree.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} root
# @return {Boolean}
def is_symmetric(root)
    return check_symmetric(root, root)
end

def check_symmetric(root, root2)
    return (root.nil? && root2.nil?) if root.nil? || root2.nil?
    return root.val == root2.val && check_symmetric(root.left, root2.right) && check_symmetric(root.right, root2.left)
end

File: javascript/easy/symmetricTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    return checkSymmmetric(root, root);
};

var checkSymmmetric = function(root1, root2) {
    if(root1 == null || root2 == null) {
        return (root1 == null && root2 == null);
    }

    return (root1.val == root2.val) && checkSymmmetric(root1.left, root2.right) && checkSymmmetric(root1.right, root2.left);
};
